src/pip/_internal/utils/logging.py

```diff
@@ -118,18 +122,32 @@ def get_message_start(self, formatted, levelno):
 
         return 'ERROR: '
 
+    if PY2:
+        # Compatibility with default_time_format and default_msec_format from
+        # Python >= 3.3.
+        default_msec_format = '%s,%03d'
+
+        def formatTime(self, record, datefmt=None):
+            ct = self.converter(record.created)
+            if datefmt:
+                s = time.strftime(datefmt, ct)
+            else:
+                t = time.strftime(self.default_time_format, ct)
+                s = self.default_msec_format % (t, record.msecs)
+            return s
```

Comment:
Hmm, this was the concern I expressed here and that @pfmoore seemed to agree with: https://github.com/pypa/pip/issues/6587#issuecomment-500651566

For comparison, how simple could the PR look if only Python versions that have the feature out of the box got the enhancement?

Reply:
Also, would it solve things just to use the logging module's default `datefmt`? It looks like that code path uses milliseconds even in Python 2 when I checked.

Reply:
Using `default_time_format` only for Python 3 would require at least 1 check in `format` and several in the tests.  updated the PR with a simpler approach that still works for both 2 and 3 and removes this block.

Reply:
Read my second comment, too. I don’t see a reply to that.

Reply:
Regarding `datefmt`, I see it passed to `formatTime` by the `Formatter.format` method only if we use `%(asctime)` in the record format ([1](https://github.com/python/cpython/blob/66d47da86aff15be34adbec02596bb3188684c0d/Lib/logging/__init__.py#L666)) and if it is passed to `formatTime` and is truthy then it bypasses the millisecond formatting ([2](https://github.com/python/cpython/blob/66d47da86aff15be34adbec02596bb3188684c0d/Lib/logging/__init__.py#L603)). So I don't think it will work for us here.

Reply:
I'm not sure if you understood my suggestion. I was suggesting changing `self.formatTime(record, "%Y-%m-%dT%H:%M:%S")` to `self.formatTime(record)` in order to use the logging module's default format.

Reply:
Ah, you're right I misunderstood. I prefer the current format since it is standard and if parsing would be easier to `split(1)` for each line of the log records. When we drop 2 support this can be cleaned up as in the initial revision (with `default_time_format`) while preserving the same format, but I suspect there will be a lot more refactoring in a lot more places when that happens.